code_generation/printer.cpp

```diff
@@ -464,8 +464,8 @@ QString Printer::Private::classImplementation(const Class &classObject, bool nes
 
     // Generate nested class functions
     if (!classObject.nestedClasses().isEmpty()) {
-        foreach (Class nestedClass, classObject.nestedClasses()) {
-            code += classImplementation(nestedClass, true);
+        for (const Class &nested : classObject.nestedClasses()) {
```

Comment:
This will detach / deep-copy. However you can't use qAsConst on a temporary. So this requires nestedClasses() to go into a local variable. I know, this sucks. Although here, the local var could also be used for the if() just above.